frontend/apps/reader/modules/readerlink.lua

```diff
@@ -688,62 +689,50 @@ end
 
 function ReaderLink:onGoToExternalLink(link_url)
     local text
```

Comment:
You're populating `self.alt_handlers_buttons` in here (previously, we were creating a `local alt_handlers_buttons` ), so on each external link, you're trashing and recreating most of the keys. This should probably better dont in :init(), like ReaderHighlight:init() does.

Reply:
I think the menu gets trashed and recreated because the options that are populated will vary according to the type of external link. For instance: if the link is a wiki link, the menu will show a wiki option. Maybe there's a clever way to keep it efficient, but my PR is just concerned with being able to add more buttons to the menu from outside of the module

Reply:
> my PR is just concerned with being able to add more buttons to the menu from outside of the module

I get it, but there must be a way to do this more cleanly :)
Either continue to get the table local, and only register your external buttons in a self.something - and add them to the local.
Or (my preference) build & keep in :init() a self.all_buttons with each a `should_add(link_url)` and/or `is_enabled(link_url)` functions, and have your plugin add them to that. Then in this function, just call these methods to build the local button table.